controllers/chat/moveToAnon.js

```diff
@@ -0,0 +1,123 @@
+const {User, ChatAnonUserInfo} = require('../../databases/models');
+const {StreamChat} = require('stream-chat');
+const ErrorResponse = require('../../utils/response/ErrorResponse');
+const {responseError, responseSuccess} = require('../../utils/Responses');
+const {CHANNEL_TYPE} = require('../../helpers/constants');
+const UsersFunction = require('../../databases/functions/users');
+
+/**
+ *
+ * @param {import("express").Request} req
+ * @param {import("express").Response} res
+ */
+const moveToAnon = async (req, res) => {
+  const {
+    targetUser,
+    anon_user_info_color_code,
+    anon_user_info_color_name,
+    anon_user_info_emoji_code,
+    anon_user_info_emoji_name
+  } = req.body;
+
+  let members = [targetUser];
+  if (!members.includes(req.userId)) members.push(req.userId);
+
+  const client = StreamChat.getInstance(process.env.API_KEY, process.env.SECRET);
+  try {
+    /**
+     * @type {import('stream-chat').OwnUserResponse}
+     */
+    const user = {
+      name: `Anonymous ${anon_user_info_emoji_name}`,
+      id: req.userId,
+      image: '',
+      username: `Anonymous ${anon_user_info_emoji_name}`
+    };
+    await client.connectUser(user, req.token);
+
+    if (client.user.name !== `Anonymous ${anon_user_info_emoji_name}`) {
+      await client.upsertUser({id: req.userId, name: `Anonymous ${anon_user_info_emoji_name}`});
+    }
+
+    const channel = client.channel('messaging', {members});
+    const createdChannel = await channel.create();
+
+    try {
+      await channel.updatePartial({
+        set: {
+          channel_type: CHANNEL_TYPE.ANONYMOUS,
+          anon_user_info_color_code,
+          anon_user_info_color_name,
+          anon_user_info_emoji_code,
+          anon_user_info_emoji_name
+        }
+      });
+    } catch (e) {
+      console.log(e);
+    }
+
```

Comment:
Move this after
`targetUserModel.push`

And add better_channel_members: all targets along with their respective anon info to the channel update partial param.
`
channel.updatePartial({
set: {
...
better_channel_member: [user1, user2] //with their own anon info if anonymous
...
}
})
`